### PR TITLE
[Scripting] Add timeout constraints for Ivy resolvers

### PR DESCRIPTION
**Pull request by ileasile**

### CODE REVIEW

**Review by ligee — CHANGES_REQUESTED** [review]
This looks like a too narrow feature to me.
Isn't it possible to set timeouts like this in some sort of global config?
I just wonder, the main-kts is not a general solution for all cases of the ivy resolving scenarios, more as a generally useful thing, so if it is possible to set the timeouts elsewhere, I would not pollute the functionality with such nuances. 
Moreover, sooner or later I'd like to drop the ivy resolver altogether, and at that moment we'll need to support all features we embedded here in the maven one, and without trying to implement it now, we cannot be sure that it is exposed in the same way there.
So, I suggest dropping it, if the functionality is easily accessible elsewhere (imho fiddling with timeouts is usually a corner case anyway). And if there is a good reason for that, please explain and we will discuss it and maybe find a better approach.

File: libraries/tools/kotlin-main-kts/src/org/jetbrains/kotlin/mainKts/impl/ivy.kt [hunk 1]
```diff
@@ -52,7 +51,7 @@ class IvyResolver : ExternalDependenciesResolver {
                 resolveArtifact(
                     artifactId[0], artifactId[1], artifactId[2],
                     if (artifactId.size > 3) artifactId[3] else null,
-                    if (artifactType.isNotEmpty()) artifactType else null,
+                    artifactType.ifEmpty { null },
```
Comment (ligee) [inline libraries/tools/kotlin-main-kts/src/org/jetbrains/kotlin/mainKts/impl/ivy.kt:54]: I would prefer not to have irrelevant changes in such commits. It makes history analysis more difficult. If you want to suggest such refactorings, please submit a separate PR, or at least in a separate commit.

**Comment by ileasile** [comment]
Arguments seem reasonable to me
